src/sagemaker_container_support/environment.py

```diff
@@ -0,0 +1,607 @@
+                                'validation': {
+                                    'TrainingInputMode': 'File',
+                                    'S3DistributionType': 'FullyReplicated',
+                                    'RecordWrapperType': 'None'
+                                }}
+
+                                You can find more information about /opt/ml/input/config/inputdataconfig.json here:
+                                https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-training-algo.html#your-algorithms-training-algo-running-container-inputdataconfig
+    """
+    return read_json(os.path.join(INPUT_CONFIG_PATH, INPUT_DATA_CONFIG_FILE))
+
+
+def channel_path(channel):  # type: (str) -> str
+    """ Returns the directory containing the channel data file(s) which is:
+    - <self.base_dir>/input/data/<channel>
+
+    For more information about channels: https://docs.aws.amazon.com/sagemaker/latest/dg/API_Channel.html
+
+    Returns:
+        (str) The input data directory for the specified channel.
+    """
+    return os.path.join(INPUT_DATA_PATH, channel)
+
+
+def gpu_count():  # type: () -> int
+    """The number of gpus available in the current container.
+
+    Returns:
+        (int): number of gpus available in the current container.
+    """
+    try:
+        cmd = shlex.split('nvidia-smi --list-gpus')
+        output = subprocess.check_output(cmd).decode('utf-8')
+        return sum([1 for x in output.split('\n') if x.startswith('GPU ')])
+    except (OSError, subprocess.CalledProcessError):
+        logger.warning('No GPUs detected (normal if no gpus installed)')
+        return 0
+
+
+def cpu_count():  # type: () -> int
+    """The number of cpus available in the current container.
+
+    Returns:
+        (int): number of cpus available in the current container.
+    """
+    return multiprocessing.cpu_count()
+
+
+class Environment(object):
+    """Provides access to aspects of the training environment relevant to training jobs, including
+    hyperparameters, system characteristics, filesystem locations, environment variables and configuration settings.
+
+    Example on how a script can use training environment:
+        ```
+        >>>import sagemaker_container_support as cs
+        >>>env = cs.Environment.create()
+
+        get the path of the channel 'training' from the inputdataconfig.json file
+        >>>training_dir = env.channel_input_dirs['training']
+
+        get a the hyperparameter 'training_data_file' from hyperparameters.json file
+        >>>file_name = env.hyperparameters['training_data_file']
+
+        # get the folder where the model should be saved
+        >>>model_dir = env.model_dir
+
+        >>>data = np.load(os.path.join(training_dir, training_data_file))
+
+        >>>x_train, y_train = data['features'], keras.utils.to_categorical(data['labels'])
+
+        >>>model = ResNet50(weights='imagenet')
+
+        unfreeze the model to allow fine tuning
```

Comment:
"unfreeze the model to allow fine tuning" what does this mean? and what are ... for (if this is a TODO, can you add "TODO"?